src/main/scala/beam/utils/scenario/ScenarioLoaderHelper.scala

```diff
@@ -0,0 +1,166 @@
+      val planCount = person.getPlans.size()
+      if (planCount == 0) {
+        scenario.getPopulation.removePerson(person.getId)
+      }
+    }
+
+    outputDirMaybe.foreach { path =>
+      if (planErrors.isEmpty) logger.info("No 'snap location' error to report for scenario plans.")
+      else SnapCoordinateUtils.writeToCsv(s"$path/${CsvFile.Plans}", planErrors)
+    }
+
+    val validPeople: Set[Id[Person]] = scenario.getPopulation.getPersons.values().asScala.map(_.getId).toSet
+
+    val households: List[Household] = scenario.getHouseholds.getHouseholds.values().asScala.toList
+    households.par.foreach { household =>
+      val members = household.getMemberIds.asScala.toSet
+      val validMembers = validPeople.intersect(members)
+
+      if (validMembers.isEmpty) {
+        scenario.getHouseholds.getHouseholdAttributes.removeAllAttributes(household.getId.toString)
+        scenario.getHouseholds.getHouseholds.remove(household.getId)
+      } else if (validMembers != members) {
+        val updatedHousehold = createHouseholdWithGivenMembers(household, validMembers.toList)
+        scenario.getHouseholds.getHouseholds.replace(household.getId, updatedHousehold)
+      }
+    }
+
+    val householdErrors: ListBuffer[ErrorInfo] = ListBuffer()
+
+    val householdsWithMembers: List[Household] = scenario.getHouseholds.getHouseholds.values().asScala.toList
+    householdsWithMembers.par.foreach { household =>
+      val householdId = household.getId.toString
+      val attr = scenario.getHouseholds.getHouseholdAttributes
+      val locationX = attr.getAttribute(householdId, "homecoordx").asInstanceOf[Double]
+      val locationY = attr.getAttribute(householdId, "homecoordy").asInstanceOf[Double]
+      val planCoord = new Coord(locationX, locationY)
+
+      snapLocationHelper.computeResult(planCoord) match {
+        case Right(splitCoord) =>
+          attr.putAttribute(householdId, "homecoordx", splitCoord.getX)
+          attr.putAttribute(householdId, "homecoordy", splitCoord.getY)
+        case Left(error @ Error.OutOfBoundingBoxError) =>
+          household.getMemberIds.asScala.toList.foreach(personId => scenario.getPopulation.getPersons.remove(personId))
+          scenario.getHouseholds.getHouseholds.remove(household.getId)
+          householdErrors.append(
+            ErrorInfo(
+              householdId,
+              Category.ScenarioHousehold,
+              error,
+              planCoord.getX,
+              planCoord.getY
+            )
+          )
+        case Left(error @ Error.R5SplitNullError) =>
+          household.getMemberIds.asScala.toList.foreach(personId => scenario.getPopulation.getPersons.remove(personId))
+          scenario.getHouseholds.getHouseholds.remove(household.getId)
+          householdErrors.append(
+            ErrorInfo(
+              householdId,
+              Category.ScenarioHousehold,
+              error,
+              planCoord.getX,
+              planCoord.getY
+            )
```

Comment:
These 2 error cases could be collapsed into a single one.

Reply:
Do you mean something like this,

```
case Left(error) =>
  householdErrors.append(
    ErrorInfo(
      householdId,
      Category.ScenarioHousehold,
      error,
      planCoord.getX,
      planCoord.getY
    )
  )
  error match {
    case Error.OutOfBoundingBoxError =>
      household.getMemberIds.asScala.toList.foreach(personId => scenario.getPopulation.getPersons.remove(personId))
      scenario.getHouseholds.getHouseholds.remove(household.getId)
    case Error.R5SplitNullError =>
      household.getMemberIds.asScala.toList.foreach(personId => scenario.getPopulation.getPersons.remove(personId))
      scenario.getHouseholds.getHouseholds.remove(household.getId)
  }
```

Reply:
No, just
```
case Left(error) =>
          household.getMemberIds.asScala.toList.foreach(personId => scenario.getPopulation.getPersons.remove(personId))
          scenario.getHouseholds.getHouseholds.remove(household.getId)
          householdErrors.append(
            ErrorInfo(
              householdId,
              Category.ScenarioHousehold,
              error,
              planCoord.getX,
              planCoord.getY
            )
          )
```

Reply:
Oh, man! What am I thinking!? I think I need coffee. I'll make the changes. Sorry for asking stupid questions on a plain simple change. :pray: